catalog/core/catalog-core-solr/src/test/java/ddf/catalog/source/solr/provider/SolrProviderUpdate.java

```diff
@@ -0,0 +1,755 @@
+          // Thumbnail
+          assertTrue(Arrays.equals(newCard.getThumbnail(), oldCard.getThumbnail()));
+
+          break;
+        default:
+          Assert.fail("Expecting one or the other of the updated records.");
+          break;
+      }
+    }
+  }
+
+  @Test
+  public void testUpdateWithNullThumbnail() throws Exception {
+    verifyAttributeUpdate(Metacard.THUMBNAIL, null);
+  }
+
+  @Test
+  public void testUpdateWithNullExpirationDate() throws Exception {
+    verifyAttributeUpdate(Metacard.EXPIRATION, null);
+  }
+
+  @Test
+  public void testUpdateWithNullContentType() throws Exception {
+    verifyAttributeUpdate(Metacard.CONTENT_TYPE, null);
+  }
+
+  @Test
+  public void testUpdateWithEmptyContentType() throws Exception {
+    verifyAttributeUpdate(Metacard.CONTENT_TYPE, "");
+  }
+
+  @Test
+  public void testUpdateWithNullContentTypeVersion() throws Exception {
+    verifyAttributeUpdate(Metacard.CONTENT_TYPE_VERSION, null);
+  }
+
+  @Test
+  public void testUpdateWithEmptyContentTypeVersion() throws Exception {
+    verifyAttributeUpdate(Metacard.CONTENT_TYPE_VERSION, "");
+  }
+
+  @Test
+  public void testUpdateWithNewResourceUri() throws Exception {
+    String uri1 = "http://youwillfindme.com/here/now";
+    verifyAttributeUpdate(Metacard.RESOURCE_URI, uri1);
+  }
+
+  private void verifyAttributeUpdate(String attributeName, Serializable updatedValue)
+      throws Exception {
+    deleteAll();
+    Calendar calendar = Calendar.getInstance();
+    calendar.setTimeInMillis(0);
+
+    // CREATE
+    MockMetacard metacard1 =
+        new MockMetacard(Library.getFlagstaffRecord(), MetacardImpl.BASIC_METACARD, calendar);
+    metacard1.setResourceURI(new URI("http://youwillfindme.com/here"));
+    create(metacard1);
+
+    // UPDATE
+    MockMetacard updatedMetacard1 =
+        new MockMetacard(Library.getTampaRecord(), MetacardImpl.BASIC_METACARD, calendar);
+    updatedMetacard1.setAttribute(attributeName, updatedValue);
+    UpdateResponse updateResponse = update(metacard1.getId(), updatedMetacard1);
+
+    // VERIFY
+    assertEquals(
+        "Testing Update operation: ", ONE_HIT, updateResponse.getUpdatedMetacards().size());
+
+    Metacard newCard = updateResponse.getUpdatedMetacards().get(0).getNewMetacard();
+    Metacard oldCard = updateResponse.getUpdatedMetacards().get(0).getOldMetacard();
+
+    Serializable oldValue = null;
+    if (oldCard.getAttribute(attributeName) != null) {
+      oldValue = oldCard.getAttribute(attributeName).getValue();
+    }
+    assertThat(oldValue, is(metacard1.getAttribute(attributeName).getValue()));
+
+    Serializable newValue = null;
+    if (newCard.getAttribute(attributeName) != null) {
+      newValue = newCard.getAttribute(attributeName).getValue();
+    }
+    assertThat(newValue, is(updatedValue));
+  }
+
+  @Test
+  public void testNullUpdate() throws IngestException {
+    UpdateResponse updateResponse =
+        provider.update(
+            new UpdateRequest() {
```

Comment:
❗️ Should we extract a class that takes in a few parameters instead of creating those new anonymous inner classes everywhere?